### ISSUE TITLE
mosquitto_subscribe() does not work in V1.6.6

### ISSUE THREAD

**Issue by LiamFry** [eclipse/mosquitto#1422]
Simple _publish_ (as `p.c`) and _subscribe_ (as `s.c`) programs (in C). The publisher works fine. The subscriber seems to never subscribe. The `mosqOnSubscribe()` callback is never called and incoming messages are never delivered to `mosqOnMessage()`.
I added some delays (`usleep()`) bc. I wasn't sure if it were a timing issue. (It's not.)

Simply compile as `gcc -o p p.c -lmosquitto` etc.

I noticed the provided `mosquitto_sub` command line utility uses `mosquitto_subscribe_v5()` -- see my earlier issue about 'where are the V5 docs.'

[Archive.zip](https://github.com/eclipse/mosquitto/files/3641830/Archive.zip)

**Comment by karlp** [comment]
this uses loop_start, so possibly seeing the same issue as https://github.com/eclipse/mosquitto/issues/1415 and https://github.com/eclipse/mosquitto/issues/1385

**Comment by ralight** [comment]
If you are able to try out the `fixes` branch and report back I would appreciate it.

```
git clone https://github.com/eclipse/mosquitto
cd mosquitto
git checkout fixes
make
```

**Comment by ralight** [comment]
This is now fixed in 1.6.7.

**Comment by LiamFry** [comment]
Apologies; I'm late to comment. I've been beating the crap out of my application [with 1.6.7] and all is great!
